src/sentry/eventstore/base.py

```diff
@@ -0,0 +1,82 @@
+from __future__ import absolute_import
+
+from sentry import nodestore
+from sentry.utils.services import Service
+
+
+class EventStorage(Service):
+    __all__ = (
+        'minimal_columns',
+        'full_columns',
+        'get_event_by_id',
+        'get_events',
+        'get_prev_event_id',
+        'get_next_event_id',
+        'bind_nodes',
+    )
+
+    # The minimal list of columns we need to get from snuba to bootstrap an
+    # event. If the client is planning on loading the entire event body from
+    # nodestore anyway, we may as well only fetch the minimum from snuba to
+    # avoid duplicated work.
+    minimal_columns = [
+        'event_id',
+        'group_id',
+        'project_id',
+        'timestamp',
+    ]
+
+    # A list of all useful columns we can get from snuba.
+    full_columns = minimal_columns + [
+        'culprit',
+        'location',
+        'message',
+        'platform',
+        'title',
+        'type',
+
+        # Required to provide snuba-only tags
+        'tags.key',
+        'tags.value',
+
+        # Required to provide snuba-only 'user' interface
+        'email',
+        'ip_address',
+        'user_id',
+        'username',
+    ]
+
+    def get_event_by_id(self, project_id, event_id, cols):
+        raise NotImplementedError
+
+    def get_events(self, start, end, cols, conditions, filter_keys, orderby, limit, offset):
+        raise NotImplementedError
+
+    def get_next_event_id(self, event, conditions, filter_keys):
+        raise NotImplementedError
+
+    def get_prev_event_id(self, event, conditions, filter_keys):
+        raise NotImplementedError
+
+    def bind_nodes(self, object_list, *node_names):
```

Comment:
Is `node_names` ever something other than `data`? The existing `bind_nodes` method is always called with `data` which makes me believe this parameter could have a default value or not be a parameter at all.

Reply:
It's not (except for https://github.com/getsentry/sentry/blob/master/src/sentry/web/frontend/error_page_embed.py#L170 which I believe is an error).

Reply:
I simplified bind_nodes to have a single node_name which defaults to `data`